contrib/go-redis/redis/redis.go

```diff
@@ -139,29 +134,30 @@ func commandsToString(cmds []redis.Cmder) string {
 
 // WithContext sets a context on a Client. Use it to ensure that emitted spans have the correct parent.
 func (c *Client) WithContext(ctx context.Context) *Client {
-	c.mu.Lock()
-	c.ctx = ctx
-	c.mu.Unlock()
-	return c
+	newClient := &Client{
```

Comment:
`newClient` sounds like a constructor function. I like what the downstream package uses: `clone`. Other better options: `cc` (client copy), `cp` (copy).

Reply:
`clone` is good.